### PR TITLE
bugfix: add err handle for obj cache.

### PR DESCRIPTION
**Pull request by moonming**

### CODE REVIEW

File: lua/apisix/core/lrucache.lua [hunk 1]
```diff
@@ -26,10 +27,16 @@ local function global_lru(key, version, create_obj_fun, ...)
...
-    obj._cache_ver = version
-    global_lrus:set(key, obj, GLOBAL_TTL)
-    return obj
+    local err
+    obj, err = create_obj_fun(...)
+    if type(obj) == 'table' then
+        obj._cache_ver = version
+        global_lrus:set(key, obj, GLOBAL_TTL)
+    else
+        log.error('failed to call create_obj_fun in global_lru()')
```
Comment (membphis) [inline lua/apisix/core/lrucache.lua]: it should be: only support to cache Lua table object.

File: lua/apisix/core/lrucache.lua [hunk 2]
```diff
@@ -49,11 +56,16 @@ local function _plugin(count, ttl, plugin_name, key, version, create_obj_fun,
...
-    obj = create_obj_fun(...)
-    obj._cache_ver = version
-    lru_global:set(key, obj, ttl)
+    local err
+    obj, err = create_obj_fun(...)
+    if type(obj) == 'table' then
+        obj._cache_ver = version
+        lru_global:set(key, obj, ttl)
+    else
+        log.error('failed to call create_obj_fun in _plugin()')
```
Comment (membphis) [inline lua/apisix/core/lrucache.lua]: ditto

File: lua/apisix/core/lrucache.lua (hunk 1 above)
Comment (membphis) [reply lua/apisix/core/lrucache.lua]: I think we shouldn't write any error logs here, just return it.

Comment (membphis) [reply lua/apisix/core/lrucache.lua]: @moonming 

$ lj-releng lua/apisix/core/*.lua
lua/apisix/core/lrucache.lua: 0.1 (0.1)
ERROR: lua/apisix/core/lrucache.lua: line 32: getting the Lua global "type"
ERROR: lua/apisix/core/lrucache.lua: line 61: getting the Lua global "type"
Checking line length exceeding 80...
lua/apisix/core/lrucache.lua:36:        log.error('failed to call create_obj_fun in global_lru(), only support to cache Lua table object.')
lua/apisix/core/lrucache.lua:65:        log.error('failed to call create_obj_fun in _plugin(), only support to cache Lua table object.')

Comment (moonming) [reply lua/apisix/core/lrucache.lua]: better add in travis

**Pull request merged**